ts/src/profilers/time-profiler.ts

```diff
@@ -35,10 +35,20 @@ export class TimeProfiler {
    * @param durationMillis - time in milliseconds for which to collect profile.
    */
   async profile(durationMillis: number): Promise<perftools.profiles.IProfile> {
+    // Node.js contains an undocumented API for reporting idle status to V8.
+    // This lets the profiler distinguish idle time from time spent in native
+    // code. Ideally this should be default behavior. Until then, use the
+    // undocumented API.
+    // See https://github.com/nodejs/node/issues/19009#issuecomment-403161559.
+    // tslint:disable-next-line no-any
+    (process as any)._startProfilerIdleNotifier();
```

Comment:
Does this API exist on all Node.js versions? Should this call and the call below check whether the API exists to avoid failing the profiling if it does not?

Reply:
It exists in all versions supported by this module.